### PR TITLE
Hot add test fixing

### PR DESCRIPTION
**Pull request by psav**
* Fix finalizer
* Fix Dialog create
* Fix locator for Instance Fields
* Allow Class to be created with Schema
* Fix random vm name for hotadd test
* Fix input -> Input() locator

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 76ae61c760b0795b122bc5532cbed246d934eed3:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 46f4ed04591693d493cfaca90b1c8f2cad371bd8:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/automate/test_vmware_methods.py [hunk 1]
```diff
@@ -41,16 +41,21 @@ def testing_group(request):
...
         provider_crud=provider_crud,
         template_name=provisioning["template"]
     )
-    request.addfinalizer(lambda: vm.remove_from_cfme())
-    request.addfinalizer(lambda: vm.delete_from_provider())
+
+    def _finalize():
+        vm.delete_from_provider()
+        if vm.does_vm_exist_in_cfme():
+            vm.remove_from_cfme()
```
Comment (seandst) [inline cfme/tests/automate/test_vmware_methods.py]: remove_from_cfme should just do this check, I think.

File: cfme/tests/automate/test_vmware_methods.py [hunk 2]
```diff
@@ -105,7 +116,7 @@ def _get_disk_count():
             properties=("Datastore Allocation Summary", "Number of Disks")).strip())
     original_disk_count = _get_disk_count()
     toolbar.select(testing_group.text, button.text)
-    fill("input#size", "1")
+    fill(Input("size"), "1")
```
Comment (seandst) [inline cfme/tests/automate/test_vmware_methods.py]: Any chance of using a real locator here, instead of something buried in a test? :)

Comment (mfalesni) [inline cfme/tests/automate/test_vmware_methods.py]: The ids of the inputs are generated based on the values you provide to the dialog.

**Pull request merged**